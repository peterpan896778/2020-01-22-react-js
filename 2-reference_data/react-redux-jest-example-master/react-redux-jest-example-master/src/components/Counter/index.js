import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'src/components';

import {
  ButtonsContainer,
  CounterContainer,
} from './styles';

const Counter = ({ increment, decrement, value }) => (
  <div>
    <ButtonsContainer>
      <Button
        label="Decrement"
        type="danger"
        onClick={decrement}
      />
      <Button
        label="Increment"
        type="success"
        onClick={increment}
      />
    </ButtonsContainer>
    <CounterContainer>
      <span>Counter: {value}</span>
    </CounterContainer>
  </div>
);

Counter.defaultProps = {
  decrement: () => {},
  increment: () => {},
  value: 0,
};

Counter.propTypes = {
  /** Counter decrement funcion */
  decrement: PropTypes.func,
  /** Counter increment funcion */
  increment: PropTypes.func,
  /** Counter value **/
  value: PropTypes.number,
};

export default Counter;
