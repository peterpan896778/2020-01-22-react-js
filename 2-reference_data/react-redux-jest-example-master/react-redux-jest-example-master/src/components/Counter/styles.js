import styled from 'styled-components';

export const ButtonsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  justify-content: space-around;
  margin: auto;
  margin-top: 20px;
  max-width: 260px;
`;

export const CounterContainer = styled.div`
  font-family: 'Open Sans', sans-serif;
  text-align: center;
  margin-top: 20px;
`;
