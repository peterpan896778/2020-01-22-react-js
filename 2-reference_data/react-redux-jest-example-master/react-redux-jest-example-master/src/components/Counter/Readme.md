If you want to use a Counter you have to import from `src/components`:
```javascript
import { Counter } from 'src/components';
```

Counter without value:

    <Counter />

Counter with value:

    <Counter value="30" />
