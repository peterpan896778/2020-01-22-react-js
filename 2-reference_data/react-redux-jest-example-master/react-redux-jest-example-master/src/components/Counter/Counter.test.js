import React from 'react';
import { mount, shallow } from 'enzyme';
import Counter from '.';

test('<Counter /> mounts without crashing', () => {
  const tree = mount(
    <Counter />,
  );
  expect(tree).toMatchSnapshot();
});

test('<Counter /> renders', () => {
  const tree = shallow(
    <Counter />,
  );
  expect(tree).toMatchSnapshot();
});

test('<Counter /> renders with value as prop', () => {
  const tree = shallow(
    <Counter value={30} />,
  );
  expect(tree).toMatchSnapshot();
});
