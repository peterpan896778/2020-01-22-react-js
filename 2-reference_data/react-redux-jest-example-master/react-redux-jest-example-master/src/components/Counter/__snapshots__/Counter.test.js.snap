// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> mounts without crashing 1`] = `
<Counter
  decrement={[Function]}
  increment={[Function]}
  value={0}
>
  <div>
    <styled.div>
      <div
        className="sc-htpNat iAWPfp"
      >
        <Button
          label="Decrement"
          onClick={[Function]}
          type="danger"
        >
          <styled.button
            label="Decrement"
            onClick={[Function]}
            type="danger"
          >
            <button
              className="sc-bdVaJa bVdEId"
              label="Decrement"
              onClick={[Function]}
              type="danger"
            >
              Decrement
            </button>
          </styled.button>
        </Button>
        <Button
          label="Increment"
          onClick={[Function]}
          type="success"
        >
          <styled.button
            label="Increment"
            onClick={[Function]}
            type="success"
          >
            <button
              className="sc-bdVaJa hjTpqZ"
              label="Increment"
              onClick={[Function]}
              type="success"
            >
              Increment
            </button>
          </styled.button>
        </Button>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="sc-bxivhb iPRuUZ"
      >
        <span>
          Counter: 
          0
        </span>
      </div>
    </styled.div>
  </div>
</Counter>
`;

exports[`<Counter /> renders 1`] = `
<div>
  <styled.div>
    <Button
      label="Decrement"
      onClick={[Function]}
      type="danger"
    />
    <Button
      label="Increment"
      onClick={[Function]}
      type="success"
    />
  </styled.div>
  <styled.div>
    <span>
      Counter: 
      0
    </span>
  </styled.div>
</div>
`;

exports[`<Counter /> renders with value as prop 1`] = `
<div>
  <styled.div>
    <Button
      label="Decrement"
      onClick={[Function]}
      type="danger"
    />
    <Button
      label="Increment"
      onClick={[Function]}
      type="success"
    />
  </styled.div>
  <styled.div>
    <span>
      Counter: 
      30
    </span>
  </styled.div>
</div>
`;
